### PR TITLE
Packaging

### PR DESCRIPTION
**Pull request by lsoucasse**
This work reorganizes the Janus repository so that it becomes a PyPI package.

The main change is that the Socrates radiation code is now put out of the repository for clarity. It is assumed that is is preinstalled before installing and using Janus (see new README.md).

Module sources are now in the src folder, as well as all data files (spectral files and luminosity tracks). Other utility scripts are in the tools folder. Demo scripts are in the test folder for future GH actions to be set up.

### CODE REVIEW

**Comment by timlichtenberg** [comment]
Just for confirmation: you specifically selected Harrison and Stef to review this, right? Fine for me, just to avoid confusion. :)

**Comment by lsoucasse** [comment]
> @lsoucasse Just for confirmation: you specifically selected Harrison and Stef to review this, right? Fine for me, just to avoid confusion. :)

Yes, because it is software-oriented and not model-oriented. But I am happy to add you as well.

**Comment by nichollsh** [comment]
It now installs fine. To get it working, I removed importlib from the `pyproject.toml` file, and also installed a version of setuptools packaged by conda rather than by pip. It was also missing the dependency f90nml.

With these changes, the model output looks good. Do you want me to push these to this branch, @lsoucasse ?

**Comment by lsoucasse** [comment]
> It now installs fine. To get it working, I removed importlib from the `pyproject.toml` file ([as suggested here](https://stackoverflow.com/a/72884543)), and also installed a version of setuptools packaged by conda rather than by pip. It was also missing the dependency f90nml.
> 
> With these changes, the model output looks good. Do you want me to push these to this branch, @lsoucasse ?

Yes, please do.

**Review by nichollsh — APPROVED** [review]
Looks great - works for me.
This will require corresponding changes in PROTEUS.

File: README.md [hunk 1]
```diff
@@ -14,26 +14,32 @@ https://proteus-code.readthedocs.io
...
+1. Download and install Socrates
+    * `git clone git@github.com:FormingWorlds/SOCRATES.git ./my_socrates`
+    * `cd my_socrates`
+    * `./configure`
+    * `./build-code`
+    * `source set_rad_env`
+2. Download and install Janus
+    * `git clone git@github.com:FormingWorlds/JANUS.git ./my_janus`
+    * `cd my_janus`
+    * `pip install -e .`
```
Comment (stefsmeets) [inline README.md:42]:
```suggestion
    * `git clone git@github.com:FormingWorlds/JANUS.git`
    * `cd JANUS`
    * `pip install -e .`
```

File: README.md [hunk 2]
```diff
@@ -14,26 +14,32 @@ https://proteus-code.readthedocs.io
...
+* `tests/demo_instellation.py`         - Calculate fluxes (and temperatures) for different instellations
+* `tools/`                             - Useful tools
+
+### Developer installation instructions
+1. Download and install Socrates
+    * `git clone git@github.com:FormingWorlds/SOCRATES.git ./my_socrates`
+    * `cd my_socrates`
+    * `./configure`
+    * `./build-code`
+    * `source set_rad_env`
```
Comment (stefsmeets) [inline README.md:38]:
```suggestion
    * `git clone git@github.com:FormingWorlds/SOCRATES.git`
    * `cd SOCRATES`
    * `./configure`
    * `./build-code`
    * `source set_rad_env`
```

**Review by stefsmeets — CHANGES_REQUESTED** [review]
Hi @lsoucasse , looks good, it seems to install fine with `pip install -e .` and `pip install .` on python 3.12. It's a bit difficult to test everything, because there are no tests. I tried to import some files and it seems mostly fine.

Couple of notes:

- pyproject.toml is missing metadata, like the required python version, licence information, urls, etc: https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#a-full-example 
- Missing import: f90nml, phys, seaborn
- The package is very large with a 230 MB download, and 873 MB unpacked. Nearly all of this is about the spectral files that are stored as plain text (872 MB). This is something we could think of optimizing.

**Comment by nichollsh** [comment]
I agree. Maybe we could pull the spectral files from OSF using [this library](https://osfclient.readthedocs.io/en/latest/)?

**Comment by lsoucasse** [comment]
> Hi @lsoucasse , looks good, it seems to install fine with `pip install -e .` and `pip install .` on python 3.12. It's a bit difficult to test everything, because there are no tests. I tried to import some files and it seems mostly fine.
> 
> Couple of notes:
> 
>     * pyproject.toml is missing metadata, like the required python version, licence information, urls, etc: https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#a-full-example
> 
>     * Missing import: f90nml, phys, seaborn
> 
>     * The package is very large with a 230 MB download, and 873 MB unpacked. Nearly all of this is about the spectral files that are stored as plain text (872 MB). This is something we could think of optimizing.

Thanks Stef!
* Licence information and project url are already in the toml file. What do you want specifically to add? For the minimum python version requirement do you have any suggestion @nichollsh?
* The "phys" module is actually an in-house one in janus.utils. There are some ambiguous imports I will fix. Maybe worth considering renaming this module?
* For the data management, we have to define a strategy for the whole PROTEUS platform. Maybe it is relevant to have some spectral data in the package (for basic usage, tests) but I agree it is better to have most of them stored in a dedicated platform.

**Comment by nichollsh** [comment]
The code certainly requires Python >= 3.10, because it includes some `match` statements which were only introduced in that version. I've been using 3.11 recently, but I think 3.10 is a reasonable requirement.

I agree that we should leave at least 1 spectral file in for tests, and that we should think about the rest.

**Review by stefsmeets — APPROVED** [review]
All good, I was thinking of being explicit about the licence like so:

```
[project]
license = {text = "Apache 2.0 License"}
```

This will show up nicer in `pip show` and other places, because it does not copy the entire file 😅

And also keywords which are useful for pypi:

```
[project]
keywords = ["egg", "bacon", "sausage", "tomatoes", "Lobster Thermidor"]
```

**Comment by lsoucasse** [comment]
> All good, I was thinking of being explicit about the licence like so:
> 
> ```
> [project]
> license = {text = "Apache 2.0 License"}
> ```
> 
> This will show up nicer in `pip show` and other places, because it does not copy the entire file 😅
> 
> And also keywords which are useful for pypi:
> 
> ```
> [project]
> keywords = ["egg", "bacon", "sausage", "tomatoes", "Lobster Thermidor"]
> ```

@timlichtenberg and @nichollsh, do you have any relevant keyword in mind for the Janus package? I would suggest "exoplanet" and "atmosphere".

**Pull request merged**